.github/workflows/test-and-deploy.yml

```diff
@@ -0,0 +1,163 @@
+
+      - name: Start deployment (staging)
+        id: deployment_staging
+        if: ${{ github.ref == format('refs/heads/{0}', inputs.default_branch) && inputs.deploy_staging }}
+        uses: bobheadxi/deployments@v1
+        with:
+          step: start
+          token: ${{ secrets.GITHUB_TOKEN }}
+          env: staging
+
+      - name: Create domain (staging)
+        if: ${{ github.ref == format('refs/heads/{0}', inputs.default_branch) && inputs.deploy_staging }}
+        run: npx sls create_domain --stage staging
+        env:
+          AWS_ACCESS_KEY_ID: ${{ secrets.AWS_ACCESS_KEY_ID }}
+          AWS_SECRET_ACCESS_KEY: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
+          DD_API_KEY: ${{ secrets.DD_API_KEY }}
+
+      - name: Deploy (staging)
+        if: ${{ github.ref == format('refs/heads/{0}', inputs.default_branch) && inputs.deploy_staging }}
+        run: poetry run inv deploy --stage staging
+        env:
+          AWS_ACCESS_KEY_ID: ${{ secrets.AWS_ACCESS_KEY_ID }}
+          AWS_SECRET_ACCESS_KEY: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
+          DD_API_KEY: ${{ secrets.DD_API_KEY }}
+
+      - name: Update deployment status (staging)
+        if: ${{ always() && github.ref == format('refs/heads/{0}', inputs.default_branch) && inputs.deploy_staging }}
+        uses: bobheadxi/deployments@v1
+        with:
+          step: finish
+          token: ${{ secrets.GITHUB_TOKEN }}
+          status: ${{ job.status }}
+          env: ${{ steps.deployment_staging.outputs.env }}
+          env_url: https://template-python-serverless-fastapi-staging.api.properly.ca/docs
+          deployment_id: ${{ steps.deployment_staging.outputs.deployment_id }}
+
+      - name: Start deployment (prod)
+        id: deployment_prod
+        if: ${{ github.ref == format('refs/heads/{0}', inputs.default_branch) }}
+        uses: bobheadxi/deployments@v1
+        with:
+          step: start
+          token: ${{ secrets.GITHUB_TOKEN }}
+          env: prod
+
+      - name: Create domain (prod)
+        if: ${{ github.ref == format('refs/heads/{0}', inputs.default_branch) }}
+        run: npx sls create_domain --stage prod
+        env:
+          AWS_ACCESS_KEY_ID: ${{ secrets.AWS_ACCESS_KEY_ID }}
+          AWS_SECRET_ACCESS_KEY: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
+          DD_API_KEY: ${{ secrets.DD_API_KEY }}
+
+      - name: Deploy (prod)
+        if: ${{ github.ref == format('refs/heads/{0}', inputs.default_branch) }}
+        run: poetry run inv deploy --stage prod
+        env:
+          AWS_ACCESS_KEY_ID: ${{ secrets.AWS_ACCESS_KEY_ID }}
+          AWS_SECRET_ACCESS_KEY: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
+          DD_API_KEY: ${{ secrets.DD_API_KEY }}
+          PROPERLY_PRICE_ESTIMATE_API_SECRET: ${{ secrets.PROPERLY_PRICE_ESTIMATE_API_SECRET }}
+
+      - name: Update deployment status (prod)
+        uses: bobheadxi/deployments@v1
+        if: ${{ always() && github.ref == format('refs/heads/{0}', inputs.default_branch) }}
```

Comment:
> Good question.. always() is used to run a step even if any previous steps fail. So here, even if the deployment fails, we want to update the deployment status to inform the user whether it passed or failed. If we do not include it, then if any of the previous steps fail, then this will not run.
This is the same reason why we have included it in publish test results step, so that even if some of the tests fail, we still publish the results for the user.